pkg/jtracer/jtracer.go

```diff
@@ -14,16 +14,57 @@
 
 package jtracer
 
-import "github.com/opentracing/opentracing-go"
+import (
+	"context"
+	"fmt"
+
+	"github.com/opentracing/opentracing-go"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+	"go.opentelemetry.io/otel/propagation"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+)
 
 type JTracer struct {
-	OT opentracing.Tracer
+	OT   opentracing.Tracer
+	OTEL trace.TracerProvider
 }
 
 func OT(t opentracing.Tracer) JTracer {
 	return JTracer{OT: t}
 }
 
+func OTEL(tp trace.TracerProvider) JTracer {
```

Comment:
I prefer a single `New(t opentracing.Tracer, tp trace.TracerProvider) JTracer` constructor to keep things simple; you can pass in noop variants of each tracer if you like, in tests. That means we can remove the `NoOp()` and `NoOpOtel()` funcs.

In fact, if we're exporting the `OT` `OTEL` fields, I don't think we even need a constructor func which will make things even simpler. I only see a need for it if we unexport the `JTracer` fields.

Thoughts?

Reply:
I would keep whichever ctors are currently convenient, and clean-up after the migration is completed. We'd have to do the final clean-up anyway, since we'd want to eliminate any OT references.

Reply:
having said that, I would recommend only having `New() JTracer` and let it create both as needed, instead of forcing main's here and there create those manually (and inconsistently)

Reply:
> I would recommend only having New() JTracer and let it create both as needed

That sounds reasonable. For tests, would we still need to do: `JTracer{OT: opentracing.NoopTracer{}}` (for example)?

Reply:
we should be aiming to minimize changes and churn. Tests should be able to call NoOp(), but why do they need to distinguish between NoOpOtel and OT? A single function can create JTracer with both sub-tracers initialized to noop.

Reply:
Hi, I could create `New() JTracer` constructor on top of Jtracer struct. But there is a lot of files being addressed by func Jtracer.OT().

Reply:
The question in my mind is, will there be a situation where we need both otel and ot tracers? If so, then we'll need the New(). If not, I think it's okay to keep OT() and OTEL() constructors.

Reply:
Also worth thinking about testability. Do we want to inject a mock tracer provider to assert the spans produced? Or am I overthinking things? If so, we may need to consider dependency injection. 

Thoughts?

Reply:
In order to make incremental changes, we will need a period where the OT and OTEL instrumentations coexist, meaning we'll need to use the OTEL bridge which can satisfy both OT and OTEL interfaces. So it's still a single New function that returns connected tracers.

As for testing, New can only be called in main (which we don't test), but everything else will be getting JTracer as a dependency, which can be manipulated for testing (usually by installing an exporter that accumulates the data and allows to inspect it).